app/scripts/views/mixins/signup-success-mixin.js

```diff
@@ -7,16 +7,28 @@
 define(function (require, exports, module) {
   'use strict';
 
+  var t = require('views/base').t;
+  var Constants = require('lib/constants');
 
   module.exports = {
     onSignUpSuccess: function (account) {
       var self = this;
       if (account.get('verified')) {
-        // user was pre-verified, notify the broker.
-        return self.invokeBrokerMethod('afterSignIn', account)
-          .then(function () {
-            self.navigate('signup_complete');
-          });
+        if (self.relier.get('preVerifyToken')) {
+          // User was pre-verified, notify the broker.
+          return self.invokeBrokerMethod('afterSignIn', account)
+            .then(function () {
+              self.navigate('signup_complete');
+            });
+        }
+
+        // Account already existed. There was no need to create it,
+        // so we just signed the user in instead.
+        // https://github.com/mozilla/fxa-content-server/issues/2778
+        return self.navigate('settings', {
```

Comment:
Always calling `invokeBrokerMethod('afterSignIn'...` the making the decision of which screen to redirect to also allows the OAuth flow to redirect back to the relier if the user signs in via the signup form.